virtualmachine/vsphere/vm.go

```diff
@@ -1733,10 +1800,6 @@ func getVirtualMachines(vm *VM) (map[string]mo.VirtualMachine, error) {
 	)
 	vmsInDc := make(map[string]mo.VirtualMachine)
 	vmsInCluster := make(map[string]mo.VirtualMachine)
-	// set up session to vcenter server
```

Comment:
Why have we removed setupSession()? I think even if it is already done, this should be done conditionally here. i.e. Iff session is not setup then do that.

Reply:
As discussed offline, the convention is to setup the session in methods exposed by libretto only as the resources are cleaned up after session is created. 
Creating session and cleaning resources in any of the unexposed methods will leave the caller without resources that are used to interact to vcenter.